devito/function.py

```diff
@@ -193,6 +187,40 @@ def __dtype_setup__(cls, **kwargs):
         else:
             return np.float32
 
+    @classmethod
+    def __staggered_setup__(cls, func, **kwargs):
+        """
+        For a given function and it's staggered property `staggered`
+        returns a tuple of 0 or 1 for each dimension:
+        0 if not staggered in the dimension
+        1 if staggered in the dimension
+        """
+        stagger = kwargs.get('staggered')
+        s = (lambda x: x if type(x) is tuple else (x,))(stagger)
+        if stagger is None:
+            func.is_Staggered = False
+            return tuple(0 for _ in func.indices)
+        func.is_Staggered = True
```

Comment:
no need to set anything here. ``is_Staggered`` is better off as a property such as ``return any(i == 1 for i in self.staggered``)